src/imitation/testing/expert_trajectories.py

```diff
@@ -0,0 +1,167 @@
+"""Test utilities to conveniently generate expert trajectories."""
+import math
+import pathlib
+import pickle
+import traceback
+import warnings
+from os import PathLike
+from pathlib import Path
+from typing import Sequence
+
+import huggingface_sb3 as hfsb3
+from filelock import FileLock
+from torch.utils import data as th_data
+
+from imitation.data import rollout, types, wrappers
+from imitation.policies import serialize
+from imitation.util import util
+
+
+def generate_expert_trajectories(
+    env_id: str,
+    num_trajectories: int,
+) -> Sequence[types.TrajectoryWithRew]:
+    """Generate expert trajectories for the given environment.
+
+    Note: will just pull a pretrained policy from the huggingface model hub.
+
+    Args:
+        env_id: The environment to generate trajectories for.
+        num_trajectories: The number of episodes to generate.
+
+    Returns:
+        A list of trajectories with rewards.
+    """
+    env = util.make_vec_env(
+        env_id,
+        post_wrappers=[lambda e, _: wrappers.RolloutInfoWrapper(e)],
+    )
+    trajectories = rollout.rollout(
+        serialize.load_policy("ppo-huggingface", env, env_name=env_id),
+        env,
+        rollout.make_sample_until(min_episodes=num_trajectories),
+    )
+    env.close()
+    return trajectories
+
+
+def lazy_generate_expert_trajectories(
+    cache_path: PathLike,
+    env_id: str,
+    num_trajectories: int,
+) -> Sequence[types.TrajectoryWithRew]:
+    """Generate or load expert trajectories from cache.
+
+    Args:
+        cache_path: A path to the folder to be used as cache for the expert
+            trajectories.
+        env_id: The environment to generate trajectories for.
+        num_trajectories: The number of trajectories to generate.
+
+    Returns:
+        A list of trajectories with rewards.
+    """
+    environment_cache_path = pathlib.Path(cache_path) / hfsb3.EnvironmentName(env_id)
+    environment_cache_path.mkdir(parents=True, exist_ok=True)
+
+    trajectories_path = environment_cache_path / "rollout.npz"
+
+    with FileLock(environment_cache_path / "rollout.npz.lock"):
+        try:
+            trajectories = types.load_with_rewards(trajectories_path)
+        except (FileNotFoundError, pickle.PickleError):  # pragma: no cover
+            warnings.warn(
+                "Recomputing expert trajectories due to the following error when "
+                "trying to load them:\n" + traceback.format_exc(),
+            )
+            trajectories = generate_expert_trajectories(env_id, num_trajectories)
+            types.save(trajectories_path, trajectories)
+
+    if len(trajectories) >= num_trajectories:
+        return trajectories[:num_trajectories]
+    else:
+        # If it is not enough, just throw clear the cache and generate more.
```

Comment:
"just clear the cache" would also be OK
```suggestion
        # If it is not enough, just throw away the cache and generate more.
```